glib-macros/src/variant_derive.rs

```diff
@@ -134,7 +137,7 @@ pub fn derive_variant_for_struct(
             let to_variant = quote! {
                 impl #generics #glib::ToVariant for #ident #generics {
                     fn to_variant(&self) -> #glib::Variant {
-                        #glib::Variant::tuple_from_iter(::std::array::IntoIter::new([
+                        #glib::Variant::tuple_from_iter(::std::iter::IntoIterator::into_iter([
```

Comment:
Why?

Reply:
I was getting deprecation warnings on nightly: https://doc.rust-lang.org/nightly/std/array/struct.IntoIter.html#method.new

Reply:
That seems problematic. If the crate using the macro is still in the 2018 edition then the `IntoIterator::into_iter()` will have different behaviour (and not work here). For the 2021 edition both have the same behaviour though.

Maybe add an `#[allow]` for the deprecation to the generated code here until we stop supporting the 2018 edition.

Reply:
This is still relevant as the generated code would now fail to compile with the 2018 edition

Reply:
I am not seeing it on CI, the depreciation warning is hidden by the macro anyway